packages/fxa-payments-server/src/routes/Subscriptions/Subscription.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useCallback } from 'react';
+import { useBooleanState, useCheckboxState } from '../../lib/hooks';
+import { Subscription as SubscriptionType } from '../../store/types';
+
+type SubscriptionProps = {
+  accessToken: string,
+  subscription: SubscriptionType,
+  cancelSubscription: Function,
+};
+export const Subscription = ({
+  accessToken,
+  cancelSubscription,
+  subscription: {
+    subscriptionId,
+    productName,
+    createdAt
+  },
+}: SubscriptionProps) => {
+  const [ cancelRevealed, revealCancel, hideCancel ] = useBooleanState();
+  const [ confirmationChecked, onConfirmationChanged ] = useCheckboxState();
+  const confirmCancellation = useCallback(
+    () => cancelSubscription(accessToken, subscriptionId),
+    [ accessToken, cancelSubscription, subscriptionId ]
+  );
+
+  return (
+    <div className="subscription">
+      <h3>{productName}</h3>
+      <p>{subscriptionId} - {productName} - {'' + new Date(createdAt)}</p>
+      <div>
+        {! cancelRevealed ? <>
+          <h3>Cancel subscription <button onClick={revealCancel}>Cancel...</button></h3>
+        </> : <>
+          <h3>Cancel subscription</h3>
+          <p>Cancelling means you&apos;ll no longer be able to access the product...</p>
+          <p>
+            <label>
+              <input type="checkbox" defaultChecked={confirmationChecked} onChange={onConfirmationChanged} />
+              Cancel my access and my saved information
```

Comment:
The UX spec calls for entering a password here to confirm. That seems like it's going to be a problem, as there's no way to verify the password easily here, and there's the security issue of having people enter their password on this page. I wonder if we should ask about removing that part?

Reply:
Issue #1088 is for adding password confirmation on cancellation. But yeah, I was wondering how we would implement an API for a password check here - might be okay if it's guarded by the subscriptions OAuth scope.